STRUCTURE.md

```diff
@@ -0,0 +1,123 @@
+# Structure and organization of security bindings
+
+## Introduction
+
+This document describes the strategy taken by the haskell-cryptography organisation on the subject of implementing and structuring bindings to cryptography libraries.
+
+This is a living document.
+
+## Structure
+
+### C-esque level
+
+This is the 'lower' level of the bindings. It is designed to serve as a thin
+layer over C, consisting only of `foreign import` bindings, any necessary setup
+to ensure that users do not have to deal with C linkage or bundling issues, and
+documentation of use.
+
+Specifically, the following principles apply:
+
+* Users of these libraries should never have to even think about how the
+  libraries are bundled or linked to. Any and all handling of this, for any
+  platform, must be done by the library.
+* These libraries cannot have any dependencies other than `base`.
+* Modules can only contain `foreign import` declarations.
```

Comment:
This is a tad _too_ restrictive, since we sometimes have to specify data types representing foreign data which is opaque to Haskell.

Reply:
I've removed this line